src/test/java/com/google/step/youtube/YoutubeInfoScraperTest.java

```diff
@@ -0,0 +1,109 @@
+import static org.mockito.Mockito.anyString;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Optional;
+import com.google.api.services.youtube.model.Channel;
+import com.google.api.services.youtube.model.ChannelContentDetails.RelatedPlaylists;
+import com.google.api.services.youtube.model.ChannelContentDetails;
+import com.google.api.services.youtube.model.ChannelListResponse;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.youtube.YouTubeRequestInitializer;
+import com.google.api.services.youtube.YouTube;
+import com.google.api.services.youtube.YouTube.Channels;
+import com.google.api.services.youtube.YouTube.Channels.List;
+import java.io.IOException;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public final class YoutubeInfoScraperTest {
+
+  private YoutubeInfoScraper scraper;
+  private ChannelListResponse mockResponse; 
+  private YouTube mockYouTubeClient;
+  private YouTube.Channels mockChannels;
+  private YouTube.Channels.List mockListChannels;
+
+  private static final String CHANNEL_ID = "CHANNEL_ID";
+  private static final String CHANNEL_ID_THAT_DOES_NOT_EXIST = "CHANNEL_ID_THAT_DOES_NOT_EXIST";
+  private static final String CHANNEL_ID_THAT_EXISTS = "CHANNEL_ID_THAT_EXISTS";
+  private static final String UPLOAD_ID = "UPLOAD_ID";
+  private static final String IOEXCEPTION = "IOEXCEPTION";
+  
+  @Before
+  public void setUp() throws Exception {
+      mockYouTubeClient = mock(YouTube.class);
+      mockChannels = mock(YouTube.Channels.class);
+      mockListChannels = mock(YouTube.Channels.List.class);
+      when(mockYouTubeClient.channels()).thenReturn(mockChannels);
+      when(mockChannels.list("contentDetails")).thenReturn(mockListChannels);
+      when(mockListChannels.setId(anyString())).thenReturn(mockListChannels);
+      scraper = new YoutubeInfoScraper(mockYouTubeClient);
+  }
+
+  @Test
+  public void scrapeChannelUploadPlaylist_nonExistentChannelId() throws Exception{
+      ChannelListResponse mockResponse = new ChannelListResponse();
+      when(mockListChannels.execute()).thenReturn(mockResponse.setItems(null));
+      Optional<String> actual = scraper.scrapeChannelUploadPlaylist(CHANNEL_ID_THAT_DOES_NOT_EXIST);
+      assertEquals(false, actual.isPresent());
+  }
+
+ @Test
+  public void scrapeChannelUploadPlaylist_emptyList() throws Exception{
+      ChannelListResponse mockResponse = new ChannelListResponse();
+      when(mockListChannels.execute()).thenReturn(mockResponse.setItems(Arrays.asList()));
+      Optional<String> actual = scraper.scrapeChannelUploadPlaylist(CHANNEL_ID);
+      assertEquals(false, actual.isPresent());
```

Comment:
optional: consider either using a Matcher: http://junit.sourceforge.net/javadoc/org/junit/Assert.html#assertThat(T,%20org.hamcrest.Matcher) or even better adopt https://github.com/google/truth but there's a small learning curve there and you need to add yet another dep into your pom.xml file.

If you adopt the truth library, it would replace all calls to the org.junit.Assert library.  (If your team does decide to adopt, you can do it piecemeal, but make sure each file picks one standard. don't mix truth and non-truth in any single file.)

Reply:
For me I don't I think I need this for my tests.  I showed it to the other interns and they might use it.